spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java

```diff
@@ -379,13 +380,27 @@ else if (this.autoStartup != null) {
 						instance.getContainerProperties()::setConsumerProperties);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 * @deprecated in favor of {@link #createContainer(TopicPartitionOffset[])}
+	 */
+	@Deprecated
+	@SuppressWarnings("deprecation")
+	@Override
+	public C createContainer(final Collection<org.springframework.kafka.support.TopicPartitionInitialOffset> topicPartitions) {
+		return createContainer(topicPartitions.stream()
+				.map(TopicPartitionOffset::fromTPIO)
+				.collect(Collectors.toList())
+				.toArray(new TopicPartitionOffset[0]));
+	}
+
 	@Override
-	public C createContainer(final Collection<TopicPartitionOffset> topicPartitions) {
+	public C createContainer(final TopicPartitionOffset[] topicsAndPartitions) {
 		KafkaListenerEndpoint endpoint = new KafkaListenerEndpointAdapter() {
 
 					@Override
-					public Collection<TopicPartitionOffset> getTopicPartitions() {
-						return topicPartitions;
+					public TopicPartitionOffset[] getTopicPartitionsToAssign() {
+						return topicsAndPartitions;
```

Comment:
Don't we need to return a copy from here do not allow end-user to modify an internal state of this component?